diff --git a/core/register_core_types.cpp b/core/register_core_types.cpp
index 6843d2c7cc..ef8d6d0b52 100644
--- a/core/register_core_types.cpp
+++ b/core/register_core_types.cpp
@@ -71,6 +71,7 @@
 #include "core/translation.h"
 #include "core/undo_redo.h"
 #include "core/io/redirect.h"
+#include "modules/youwin/result.h"
 
 static Ref<ResourceFormatSaverBinary> resource_saver_binary;
 static Ref<ResourceFormatLoaderBinary> resource_loader_binary;
@@ -90,6 +91,7 @@ static _JSON *_json = nullptr;
 
 static Redirect *redir = nullptr;
 static IP *ip = nullptr;
+static Safely *_safely = nullptr;
 
 static _Geometry *_geometry = nullptr;
 
@@ -215,6 +217,7 @@ void register_core_types() {
 
 	redir = Redirect::create();
 	ip = IP::create();
+	_safely = Safely::create();
 
 	_geometry = memnew(_Geometry);
 
@@ -269,6 +272,7 @@ void register_core_singletons() {
 	Engine::get_singleton()->add_singleton(Engine::Singleton("InputMap", InputMap::get_singleton()));
 	Engine::get_singleton()->add_singleton(Engine::Singleton("JSON", _JSON::get_singleton()));
 	Engine::get_singleton()->add_singleton(Engine::Singleton("Redirect", Redirect::get_singleton()));
+	Engine::get_singleton()->add_singleton(Engine::Singleton("Safely", Safely::get_singleton()));
 }
 
 void unregister_core_types() {
@@ -308,6 +312,9 @@ void unregister_core_types() {
 	if (ip) {
 		memdelete(ip);
 	}
+	if (_safely) {
+		memdelete(_safely);
+	}
 
 	ResourceLoader::finalize();
 
